### PR TITLE
New module: manage Citrix Netscaler server configuration (network/netscaler/netscaler_server) 

### PR DESCRIPTION
**Pull request by giorgos-nikolopoulos**
##### SUMMARY

This PR adds a new module for Netscaler, netscaler_server.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

lib/ansible/modules/network/netscaler/netscaler_server.py

##### ANSIBLE VERSION

```
ansible 2.4.0 (netscaler_server_module 96927db1cd) last updated 2017/06/09 15:11:23 (GMT +300)
  config file = 
  configured module search path = [u'/home/georgen/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/georgen/repos/ansible_fork/lib/ansible
  executable location = /home/georgen/repos/ansible_fork/bin/ansible
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]

```

##### ADDITIONAL INFORMATION

Unit tests are provided.
Integration tests are provided.

### CODE REVIEW

File: lib/ansible/modules/network/netscaler/netscaler_server.py [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+        domainresolveretry=dict(type='int'),
+        ipv6address=dict(
+            type='bool',
+            default=False
+        ),
+        comment=dict(type='str'),
+        td=dict(type='float'),
+    )
+
+    hand_inserted_arguments = dict(
```
Comment (Qalthos) [inline lib/ansible/modules/network/netscaler/netscaler_server.py:213]: Why are these two different dictionaries?

Comment (giorgos-nikolopoulos) [reply lib/ansible/modules/network/netscaler/netscaler_server.py:213]: The modules are initially auto generated by parsing the NITRO objects and producing the initial boilerplate code. Afterwards the manual changes take place.

To distinguish between the arguments that were auto generated and the ones added manually I have 2 dictionaries. ``module_specific_arguments`` is the auto generated dict and ``hand_inserted_arguments`` contains the manually added arguments.

**Review by ganeshrn — APPROVED** [review]

**Comment by ganeshrn** [comment]
Closing and re-opening PR to re-run CI

**Comment by ganeshrn** [comment]
@giorgos-nikolopoulos Thank you!

**Pull request merged**